src/components/ReportActionItemIOUPreview.js

```diff
@@ -0,0 +1,161 @@
+import styles from '../styles/styles';
+
+const personalDetailsPropTypes = PropTypes.shape({
+    // This is either the user's full name, or their login if full name is an empty string
+    displayName: PropTypes.string.isRequired,
+});
+
+const propTypes = {
+    // All the data of the action
+    action: PropTypes.shape(ReportActionPropTypes).isRequired,
+
+    // Is this the most recent IOU Action?
+    isMostRecentIOUReport: PropTypes.bool.isRequired,
+
+    /* Window Dimensions Props */
+    ...windowDimensionsPropTypes,
+
+    /* --- Onyx Props --- */
+    // Active IOU Report for current report
+    iou: PropTypes.shape({
+        // Email address of the manager in this iou report
+        managerEmail: PropTypes.string,
+
+        // Email address of the creator of this iou report
+        ownerEmail: PropTypes.string,
+
+        // Outstanding amount of this transaction
+        cachedTotal: PropTypes.string,
+    }),
+
+    // All of the personal details for everyone
+    personalDetails: PropTypes.objectOf(personalDetailsPropTypes).isRequired,
+
+    // Session info for the currently logged in user.
+    session: PropTypes.shape({
+        // Currently logged in user email
+        email: PropTypes.string,
+    }).isRequired,
+};
+
+const defaultProps = {
+    iou: null,
+};
+
+class ReportActionItemIOUPreview extends React.Component {
+    /**
+     * Creates IOU preview pay button.
+     *
+     * @returns {View}
+     */
+    createPayButton() {
+        return (
+            <TouchableOpacity
+                style={[styles.buttonSmall, styles.normalRadius, styles.buttonSuccess, styles.mt6]}
+            >
+                <Text
+                    style={[
+                        styles.buttonSmallText,
+                        styles.buttonSuccessText,
+                    ]}
+                >
+                    Pay
+                </Text>
+            </TouchableOpacity>
+        );
+    }
+
+    /**
+     * Creates IOU preview box view if outstanding amount is not 0.
+     *
+     * @returns {View}
+     */
+    createPreviewBox() {
+        const managerName = lodashGet(
+            this.props.personalDetails,
+            [this.props.iou.managerEmail, 'displayName'],
+            this.props.iou.managerEmail,
+        );
+        const ownerName = lodashGet(
+            this.props.personalDetails,
+            [this.props.iou.ownerEmail, 'displayName'],
+            this.props.iou.ownerEmail,
+        );
+        const managerAvatar = lodashGet(this.props.personalDetails, [this.props.iou.managerEmail, 'avatar'], '');
+        const ownerAvatar = lodashGet(this.props.personalDetails, [this.props.iou.ownerEmail, 'avatar'], '');
+        const sessionEmail = lodashGet(this.props.session, 'email', null);
+        const cachedTotal = this.props.iou.cachedTotal.replace(/[()]/g, '');
+        const amount = Number(cachedTotal.substring(1));
+
+        if (amount === 0) {
+            return null;
+        }
```

Comment:
This is not quite working as expected, we seem to keep showing the preview with an amount of 0 until logging out/in again.

Reply:
I'm now using report.hasOutstandingIOU 👍